cpp/src/arrow/builder.h

```diff
@@ -91,30 +91,31 @@ class ARROW_EXPORT ArrayBuilder {
   /// Set the next length bits to not null (i.e. valid).
   Status SetNotNull(int64_t length);
 
-  /// Allocates initial capacity requirements for the builder.  In most
-  /// cases subclasses should override and call their parent class's
-  /// method as well.
-  virtual Status Init(int64_t capacity);
+  /// \brief Ensure that at least the indicated number of elements can be appended to
+  /// the builder without a reallocation. Does not account for reallocations
+  /// that may be due to variable size data, like binary values.
+  /// \param[in] capacity the minimum number of additional array values
+  /// \return Status
+  virtual Status Resize(int64_t capacity);
 
-  /// Resizes the null_bitmap array.  In most
-  /// cases subclasses should override and call their parent class's
-  /// method as well.
-  virtual Status Resize(int64_t new_bits);
+  /// \brief Ensure that there is enough space allocated to add the indicated
```

Comment:
Reserve reserves the capacity exponentially, too. Maybe it's also worth noting, to highlight that it's not just `Resize(length_+additional_capacity)`.

Reply:
Agreed with @alendit.

Reply:
Excellent point. Added